src/linear_algebra/gram_schmidt_ortho.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2022 Jiale Miao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiale Miao, Kevin Buzzard
+-/
+
+import tactic.basic
+import algebra.big_operators.basic
+import analysis.inner_product_space.basic
+import analysis.normed_space.is_R_or_C
+
+/-!
+# Gram-Schmidt Orthogonalization and Orthonormalization
+
+In this file we introduce Gram-Schmidt Orthogonalization and Orthonormalization
+
+## Main results
+
+- `proj` : projection between two vectors in the inner product space
+- `GS`   : Gram-Schmidt Process
+- `GS_Orthogonal` : the proof that "GS" produces an orthogonal system of vectors
+- `GS_unit` : Normalized "Gram-Schmidt" (i.e each vector in this system has unit length)
+- `GS_Orthornormal` : the proof that "GS_unit" produces an orthornormal system of vectors
+-/
+
+open_locale big_operators
+
+variables (𝕜 : Type*) (E : Type*) [is_R_or_C 𝕜] [inner_product_space 𝕜 E]
+
+/-- projection in the inner product space -/
+def proj (u v : E) : E := ((inner u v) / (inner u u) : 𝕜) • u
+
```

Comment:
This definition is in the root namespace, which means that it monopolizes the name `proj` universally for all of mathlib. This is something we try to avoid doing -- to see if it makes sense for something to be in the root namespace, you can ask yourself the question "Is this the only notion in mathematics for which this name makes sense?"

Now, in this specific case, what you probably want to do is use results from `analysis/inner_product_space/projection`. Things like [orthogonal_projection_singleton](https://leanprover-community.github.io/mathlib_docs/analysis/inner_product_space/projection.html#orthogonal_projection_singleton) look a lot like what you would want here.

Reply:
Thanks a lot. I found that lemma and I have changed it